ModulemdTranslationHelpers/Utils.py

```diff
@@ -58,3 +63,94 @@ def get_translation_catalog_from_index(index, project_name):
         catalog.add(translatable_string, locations=locations)
 
     return catalog
+
+
+def get_modulemd_translations_from_catalog(catalogs, index):
+    # Nested dictionary `data` contains information from catalog like:
+    # Key_1: <module>:<stream>
+    # Value_1: Key_2: <locale>
+    #          Value_2: list of (<string_type>:<msg.string>)
+
+    data = defaultdict(defaultdict)
+
+    for catalog in catalogs:
+        for msg in catalog:
+            for location, _ in msg.locations:
+                (module_name, stream_name, string_type,
+                 profile_name) = split_location(location)
+
+                key_1 = ("{}:{}").format(module_name, stream_name)
```

Comment:
Don't make the key into a string. Python dictionaries can handle any basic type here, so it'll be easiest to use a tuple of `(module_name, stream_name)` as the key instead of `"module_name:stream_name"` which you'll have to split again later.